ledger/src/bigtable_upload_service.rs

```diff
@@ -59,11 +57,10 @@ impl BigTableUploadService {
                 break;
             }
 
-            let end_slot = block_commitment_cache
-                .read()
-                .unwrap()
-                .highest_confirmed_root()
-                .saturating_sub(LARGEST_CONFIRMED_ROOT_UPLOAD_DELAY as u64);
+            let end_slot = min(
+                max_complete_transaction_status_slot.load(Ordering::SeqCst),
+                block_commitment_cache.read().unwrap().root(),
```

Comment:
Changing to `let end_slot = block_commitment_cache.read().unwrap().root();` seems like it could be sufficient, is there a chance that we rooted a block but didn't finish storing tx metadata for it?

Reply:
Yes, since the TransactionStatusService is async, there's a chance, albeit a small one.